File: auturbo_2018_spring/auturbo_2018_spring_week3/calculator/nodes/number_generator.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

# Author: (AuTURBO) Leon Ryuwoon Jung 

import rospy
from std_msgs.msg import Int32

class NumberGenerator():
    def __init__(self):
        self.pub_number = rospy.Publisher('/number', Int32, queue_size = 1)

        self.i = 0

        loop_rate = rospy.Rate(10)
        while not rospy.is_shutdown():
            self.fnPubNumber()
            loop_rate.sleep()

    def fnPubNumber(self):
        num_for_pub = Int32()

        self.i = self.i + 1

        num_for_pub.data = self.i

        rospy.loginfo('generated number : %d', self.i)

        self.pub_number.publish(num_for_pub)

    def main(self):
        rospy.spin()

if __name__ == '__main__':
    rospy.init_node('number_generator')
    node = NumberGenerator()
    node.main()

File: auturbo_2018_spring/auturbo_2018_spring_week3/calculator/src/number_generator.cpp
// Author: (AuTURBO) Leon Ryuwoon Jung 

#include "ros/ros.h" 
#include "std_msgs/Int32.h"

class NumberGenerator
{
 public:
  NumberGenerator()
  {
    pub_number = n.advertise<std_msgs::Int32>("/number", 1);

    ros::Rate loop_rate(10);
    while (ros::ok())
    { 
      fnPubNumber();

      ros::spinOnce();
      loop_rate.sleep();
    }
  }

  void fnPubNumber()
  {
    std_msgs::Int32 num_for_pub;

    i = i + 1;

    num_for_pub.data = i;

    ROS_INFO("generated number : %d", i);

    pub_number.publish(num_for_pub);
  }

 private:
  ros::NodeHandle n;

  ros::Publisher pub_number;

  int i = 0;
};

int main(int argc, char **argv)
{
  ros::init(argc, argv, "number_generator");

  NumberGenerator numberGenerator;

  ros::spin();

  return 0;
}